tests/system_tests/test_core/test_mode_disabled_full.py

```diff
@@ -113,5 +113,16 @@ def test_disabled_no_activity(wandb_backend_spy):
         run.mark_preempting()
         run.to_html()
         run.display()
+    assert graphql_spy.total_calls == 0
+
 
+def test_disabled_mode_artifact(wandb_backend_spy):
+    gql = wandb_backend_spy.gql
+    graphql_spy = gql.Capture()
+    wandb_backend_spy.stub_gql(gql.any(), graphql_spy)
+    run = wandb.init(settings={"mode": "disabled"})
+    art = run.log_artifact(wandb.Artifact("dummy", "dummy")).wait()
+    assert art.state is None
```

Comment:
question: Wondering, do we need more precise behavior like this too (ie mimic Artifact property/function return types ~accurately)?

If yes, this'll probably get more complicated 😕 .  Also, at some point, if we need an object that fully behaves like a ("hollowed-out / no-op") Artifact, we may just need to find a way to implement it in full

Reply:
In the most ideal state in disabled mode, we check to make sure that the artifact fields and functions the user is using are valid but don't do any network calls.
Example customer usage: https://community.wandb.ai/t/how-to-deal-with-artifact-wait-when-running-in-mode-disabled/2607  
Seems to me customers just what to have an online/offline version of their code and aren't necessarily trying to validate that their code is correct.

With this PR its just a bandaid fix that allows for `run.log_artifact(artifact).wait()` to work and also allows something like `run.log_artifact(artifact).non_existent()` to "work". I'm on the end of just letting this PR through and having a longer term fix that re-writes how we do all of this mocking (SDK team also mentioned re-writing this in later date).